src/Adapter/Order/Repository/OrderRepository.php

```diff
@@ -71,4 +73,40 @@ public function get(OrderId $orderId): Order
 
         return $order;
     }
+
+    /**
+     * Get Order by cartId
+     *
+     * @param CartId $cartId
+     *
+     * @return Order|null
+     *
+     * @throws CoreException
+     */
+    public function findByCartId(CartId $cartId): ?Order
+    {
+        try {
+            $orderId = Db::getInstance()->getValue(
+                'SELECT `id_order` FROM `' . _DB_PREFIX_ . 'orders` WHERE `id_cart` = ' . (int) $cartId->getValue(),
+                false
```

Comment:
would be cleaner to rely on DBAL/QueryBuilder. smth like (:warning: adjust test the code bellow, its just pseudo code)
```
$this->connection->createQueryBuilder()
    ->select('id_order')
    ->from($this->dbPrefix . 'orders')
    ->where('id_cart = :cartId)
    ->setParameter('cartId', $cartId->getValue)
    ->execute()
    ->fetchOne()
```